### DISCUSSION TITLE
502 Bad Gateway with new ZBDongle-E and Raspberry PI 4

### DISCUSSION THREAD

**Discussion by andreasbalzer** [Koenkk/zigbee2mqtt#13734] [General]
Hi,
I thought I'd give Home Assistant a try and got myself a Raspberry Pi 4 and Sonoff ZBDongle-E. I flashed the Home Assistant OS onto the Raspberry.

I first tried using ZHA, that let me install the Dongle and search for devices, but didn't find anything.

Trying Zigbee2MQTT didn't get me that far - I am only seeing 502 Bad Gateway in UI.
Any pointers would be great. :-)

Steps:

- I installed Mosquitto broker and started it.
- I set up a user account mqtt with a password.
- I installed Zigbee2MQTT Edge (tried the other one before without luck)
- I set config:
data_path: /config/zigbee2mqtt
**socat:**
  enabled: false
  master: pty,raw,echo=0,link=/tmp/ttyZ2M,mode=777
  slave: tcp-listen:8485,keepalive,nodelay,reuseaddr,keepidle=1,keepintvl=1,keepcnt=5
  options: "-d -d"
  log: true
**mqtt:**
  server: mqtt://localhost:1883
  user: "mqtt"
  password: "" # password-here
**serial:**
  port: >-
    /dev/serial/by-id/usb-ITEAD_SONOFF_Zigbee_3.0_USB_Dongle_Plus_V2_20220712150851-if00 # got this from Settings > Hardware
  adapter: ezsp # to prevent 
**advanced:**
  channel: 15
  log_level: debug
  log_output: console

Log-tab gives
s6-rc: info: service z2m: starting
s6-rc: info: service socat: starting
s6-rc: info: service s6rc-oneshot-runner: starting
s6-rc: info: service z2m successfully started
s6-rc: info: service socat successfully started
s6-rc: info: service s6rc-oneshot-runner successfully started
s6-rc: info: service fix-attrs: starting
s6-rc: info: service fix-attrs successfully started
s6-rc: info: service legacy-cont-init: starting
s6-rc: info: service legacy-cont-init successfully started
s6-rc: info: service legacy-services: starting
s6-rc: info: service legacy-services successfully started
[14:57:45] INFO: Socat not enabled
[14:57:47] INFO: Handing over control to Zigbee2MQTT Core ...
Zigbee2MQTT:info  2022-08-28 14:57:52: Logging to console and directory: '/config/zigbee2mqtt/log/2022-08-28.14-57-52' filename: log.txt
Zigbee2MQTT:info  2022-08-28 14:57:52: Starting Zigbee2MQTT version 1.27.0-dev (commit #77bd76d)
Zigbee2MQTT:info  2022-08-28 14:57:52: Starting zigbee-herdsman (0.14.48)

Without the adapter: ezsp I get
s6-rc: info: service z2m: starting
s6-rc: info: service socat: starting
s6-rc: info: service s6rc-oneshot-runner: starting
s6-rc: info: service socat successfully started
s6-rc: info: service z2m successfully started
s6-rc: info: service s6rc-oneshot-runner successfully started
s6-rc: info: service fix-attrs: starting
s6-rc: info: service fix-attrs successfully started
s6-rc: info: service legacy-cont-init: starting
s6-rc: info: service legacy-cont-init successfully started
s6-rc: info: service legacy-services: starting
s6-rc: info: service legacy-services successfully started
[14:58:57] INFO: Socat not enabled
[14:58:59] INFO: Handing over control to Zigbee2MQTT Core ...
Zigbee2MQTT:info  2022-08-28 14:59:04: Logging to console and directory: '/config/zigbee2mqtt/log/2022-08-28.14-59-04' filename: log.txt
Zigbee2MQTT:info  2022-08-28 14:59:04: Starting Zigbee2MQTT version 1.27.0-dev (commit #77bd76d)
Zigbee2MQTT:info  2022-08-28 14:59:04: Starting zigbee-herdsman (0.14.48)
Zigbee2MQTT:error 2022-08-28 14:59:25: Error while starting zigbee-herdsman
Zigbee2MQTT:error 2022-08-28 14:59:25: Failed to start zigbee
Zigbee2MQTT:error 2022-08-28 14:59:25: Check https://www.zigbee2mqtt.io/guide/installation/20_zigbee2mqtt-fails-to-start.html for possible solutions
Zigbee2MQTT:error 2022-08-28 14:59:25: Exiting...
Zigbee2MQTT:error 2022-08-28 14:59:25: Error: Failed to connect to the adapter (Error: SRSP - SYS - ping after 6000ms)
    at ZStackAdapter.start (/app/node_modules/zigbee-herdsman/src/adapter/z-stack/adapter/zStackAdapter.ts:102:27)
    at Controller.start (/app/node_modules/zigbee-herdsman/src/controller/controller.ts:131:29)
    at Zigbee.start (/app/lib/zigbee.ts:58:27)
    at Controller.start (/app/lib/controller.ts:101:27)
    at start (/app/index.js:109:5)
[14:59:27] INFO: Handing over control to Zigbee2MQTT Core ...
Zigbee2MQTT:info  2022-08-28 14:59:32: Logging to console and directory: '/config/zigbee2mqtt/log/2022-08-28.14-59-32' filename: log.txt
Zigbee2MQTT:info  2022-08-28 14:59:32: Starting Zigbee2MQTT version 1.27.0-dev (commit #77bd76d)
Zigbee2MQTT:info  2022-08-28 14:59:32: Starting zigbee-herdsman (0.14.48)
Zigbee2MQTT:error 2022-08-28 14:59:53: Error while starting zigbee-herdsman
Zigbee2MQTT:error 2022-08-28 14:59:53: Failed to start zigbee
Zigbee2MQTT:error 2022-08-28 14:59:53: Check https://www.zigbee2mqtt.io/guide/installation/20_zigbee2mqtt-fails-to-start.html for possible solutions
Zigbee2MQTT:error 2022-08-28 14:59:53: Exiting...
Zigbee2MQTT:error 2022-08-28 14:59:53: Error: Failed to connect to the adapter (Error: SRSP - SYS - ping after 6000ms)
    at ZStackAdapter.start (/app/node_modules/zigbee-herdsman/src/adapter/z-stack/adapter/zStackAdapter.ts:102:27)
    at Controller.start (/app/node_modules/zigbee-herdsman/src/controller/controller.ts:131:29)
    at Zigbee.start (/app/lib/zigbee.ts:58:27)
    at Controller.start (/app/lib/controller.ts:101:27)
    at start (/app/index.js:109:5)
[14:59:55] INFO: Handing over control to Zigbee2MQTT Core ....

**Comment by gerardsyd** [comment]
When you run it with adapter: ezsp, Z2m is starting. 
To see the  UI, you need to add frontend: true to config file.

**Reply by andreasbalzer** [reply]
Thank you. By chance I got a log entry
Zigbee2MQTT:info  2022-09-02 02:04:09: Connecting to MQTT server at mqtt://localhost:1883
Zigbee2MQTT:debug 2022-09-02 02:04:09: Using MQTT login with username: mqtt
Zigbee2MQTT:error 2022-09-02 02:04:09: MQTT failed to connect: connect ECONNREFUSED 127.0.0.1:1883

I replaced localhost with the IPv4 and then it started working :-)
